setup.js

```diff
@@ -1,16 +1,112 @@
-const { helpers, Output, Tx, Outpoint } = require('leap-core');
-const Node = require('./client');
+const { Node, formatHostname, unspentForAddress } = require('./client');
+const fs = require('fs');
+const Web3 = require('web3');
+const bridgeAbi = require('./build/node/src/abis/bridgeAbi');
+const exitHandlerAbi = require('./build/node/src/abis/exitHandler');
+const operatorAbi = require('./build/node/src/abis/operator');
+const erc20abi = require('./erc20abi');
+
+const bip39 = require('bip39');
+const hdkey = require('ethereumjs-wallet/hdkey');
+const wallet = require('ethereumjs-wallet');
+
+const { helpers, Tx, Outpoint, Output } = require('leap-core');
+
+function getAccounts(mnemonic, num) {
+  const accounts = [];
+  for (let i=0; i<num; i++) {
+    accounts.push(getAccount(mnemonic, i));
+  }
+  return accounts;
+}
+
+function getAccount(mnemonic, id) {
+  const seed = bip39.mnemonicToSeed(mnemonic); // mnemonic is the string containing the words
+  const hdk = hdkey.fromMasterSeed(seed);
+  const addr_node = hdk.derivePath("m/44'/60'/0'/0/" + id); //m/44'/60'/0'/0/0 is derivation path for the first account. m/44'/60'/0'/0/1 is the derivation path for the second account and so on
+  const addr = addr_node.getWallet().getAddressString(); //check that this is the same with the address that ganache list for the first account to make sure the derivation is correct
+  const private_key = '0x' + addr_node.getWallet().getPrivateKey().toString('hex');
+
+  return {
+    addr: addr,
+    privKey: private_key
+  }
+}
+
+function parseConfig() {
+  const config = {};
+  fs.readFileSync('config', {encoding: 'utf8'}).toString().split('\n').forEach(line => {
+    [key, value] = line.split('=');
+    value = value.replace(/['"]+/g, '');
+    config[key] = value;
+  });
+  return config;
+}
+
+async function setup(contracts, nodes, accounts, web3) {
+  const alice = accounts[0].addr;
+  console.log(alice);
+  web3.eth.getBalance(alice).then(console.log);
+  await contracts.token.methods.approve(contracts.exitHandler.options.address, 500000000000).send({from: alice});
+  await contracts.token.methods.approve(contracts.operator.options.address, 500000000000).send({from: alice});
+
+  let slotId = 0;
+  for (let i = 0; i < nodes.length; i++) { 
+    let node = nodes[i];
+    const info = await node.web3.getValidatorInfo();
```

Comment:
This shouldn't be a part of our API, you can fetch it from tendermint directly

Reply:
But if we'll implement https://github.com/leapdao/leap-node/issues/108 it can come in handy 🤔

Reply:
I think this is very useful. It basically allows us to say to the node: please identify yourself. We definitely need this for the ethereum address, so might as well add tendermint next to it no?